src/main/bash/sdkman-init.sh

```diff
@@ -30,24 +30,67 @@ if [ -z "$SDKMAN_DIR" ]; then
 fi
 
 # infer platform
-SDKMAN_PLATFORM="$(uname)"
-if [[ "$SDKMAN_PLATFORM" == 'Linux' ]]; then
-	if [[ "$(uname -m)" == 'i686' ]]; then
-		SDKMAN_PLATFORM+='32'
-	elif [[ "$(uname -m)" == 'aarch64' ]]; then
-		SDKMAN_PLATFORM+='ARM64'
-	else
-		SDKMAN_PLATFORM+='64'
-	fi
-fi
+
+function infer_platform() {
+	local kernel
+	local machine
+	
+	kernel="$(uname -s)"
+	machine="$(uname -m)"
+	
+	case $kernel in
+	Linux)
+	  case $machine in
+	  i686)
+		echo "Linux32"
+		;;
+	  X86_64)
```

Comment:
I'm not sure if this is the case always, but in my system i see the x lowercase: x86-64. Not the end of the world cause the default case catches it, but I still think it should be lowercase to actually match the correct string

Reply:
Thanks for the feedback @pago88. Would you be able to paste your exact shell output for `uname -s` and `uname -m` please? Also, could I ask you to please drop me a dump of `neofetch` if you have it installed?

Reply:
I'm running/coding sdkman inside a docker container, so uname -s is Linux and uname -m is x86_64 (lowercase x). For what it's worth, my host is an intel mac, and uname -m also returns x86_64, lowercase x
Result of neofetch on the container:

       _,met$$$$$gg.          root@759d2e7b5100 
    ,g$$$$$$$$$$$$$$$P.       ----------------- 
  ,g$$P"     """Y$$.".        OS: Debian GNU/Linux 10 (buster) x86_64 
 ,$$P'              `$$$.     Host: BHYVE 1.0 
',$$P       ,ggs.     `$$b:   Kernel: 4.19.121-linuxkit 
`d$$'     ,$P"'   .    $$$    Uptime: 5 hours, 46 mins 
 $$P      d$'     ,    $$P    Packages: 233 (dpkg) 
 $$:      $$.   -    ,d$$'    Shell: bash 5.0.3 
 $$;      Y$b._   _,d$P'      CPU: Intel i5-5257U (2) @ 2.698GHz 
 Y$$.    `.`"Y$$$$P"'         Memory: 1274MiB / 1989MiB 
 `$$b      "-.__
  `Y$$                                                
   `Y$$.
     `$$b.
       `Y$$b.
          `"Y$b._
              `"""

Reply:
Great stuff, thanks for taking the time. I'll make that an explicit check.

Reply:
That's been fixed. Thanks!